### PR TITLE
Fix wasm credentials synchronisation

### PR DESCRIPTION
**Pull request by azgabur**
In next PR/MR I will implement the missing parts for

- credentials location: authorization
- auth method: oidc

### CODE REVIEW

File: testsuite/tests/service_mesh/auth/test_user_key_app_id_wasm_query.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    pytest.param(Service.AUTH_APP_ID_KEY, id="app-id")
+])
+def service_settings(request, service_settings):
+    """Set auth mode to user key and app-id"""
+    service_settings.update(backend_version=request.param)
+    return service_settings
+
+
+@pytest.fixture(scope="module")
+def service_proxy_settings(service_proxy_settings):
```
Comment (pehala) [inline testsuite/tests/service_mesh/auth/test_user_key_app_id_wasm_query.py:25]: This could theoretically be parametrized as well?

**Review by pehala — CHANGES_REQUESTED** [review]

File: testsuite/tests/service_mesh/auth/test_user_key_app_id_wasm_headers.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+"""
```
[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/tests/service_mesh/auth/test_user_key_app_id_wasm_headers.py]: addressed

File: testsuite/gateways/wasm/extension.py [hunk 3]
```diff
@@ -106,6 +117,64 @@ def synchronise_mapping_rules(self):
...
+                "value": []
+            }
+        ]
+
+        self.httpbin.patch("sme", self.extension_name, ops, patch_type="json")
+
+    def remove_all_credentials(self):
+        """Remove all credentials in extension"""
+        for label in ["user_key", "app_id", "app_key"]:
+            self.remove_credential(label)
```
[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/gateways/wasm/extension.py]: addressed

File: testsuite/gateways/wasm/extension.py [hunk 4]
```diff
@@ -106,6 +117,64 @@ def synchronise_mapping_rules(self):
...
+        credentials_location = config["content"]["proxy"]["credentials_location"]
+        translate = {"query": "query_string", "headers": "header", "authorization": "authorization"}
+        credentials_location = translate[credentials_location]
+
+        auth_method = config["content"]["proxy"]["authentication_method"]
+        if auth_method == Service.AUTH_USER_KEY:
+            self.add_credential("user_key", keys_obj(auth_user_key, credentials_location))
+        elif auth_method == Service.AUTH_APP_ID_KEY:
+            self.add_credential("app_id", keys_obj(auth_app_id, credentials_location))
+            self.add_credential("app_key", keys_obj(auth_app_key, credentials_location))
```
[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/gateways/wasm/extension.py]: addressed

File: testsuite/gateways/wasm/__init__.py [hunk 5]
```diff
@@ -76,6 +77,8 @@ def add_mapping_rules(self, service, rules):
     def _create_api_client(self, application, endpoint, verify, cert=None, disable_retry_status_list=None):
         ext = self.extensions[application.service["id"]]
         ext.synchronise_mapping_rules()
+        ext.synchronise_credentials()
+        sleep(10)  # wait needed for WASM to load new configuration
```
Comment (pehala) [inline testsuite/gateways/wasm/__init__.py:81]: Would something like this `ext.httpbin.deployment(f"dc/{ext.httpbin_name}").wait_for()` work too? Even though I hate the idea of redeploying httpbin due to config change, I hate the idea of `sleep(10)` more

Comment (azgabur) [reply testsuite/gateways/wasm/__init__.py]: Tried this, but sadly was not enough.

[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/gateways/wasm/__init__.py]: This works. Its not ideal solution I agree, but right now I cant think of other way. 
From my testing it prolonged test run by about 3 minutes. From about 4:30 to 7:40 runtime.

**Review by pehala — APPROVED** [review]

**Review by mdujava — APPROVED** [review]

**Pull request merged**